Add files via upload

antonychiu2 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

antonychiu2 committed:
Add files via upload

Commit:
Auto commit for SQL_Injection-5b692a17-c2d3-42fb-b097-13d1c3013a96

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/5b692a17-c2d3-42fb-b097-13d1c3013a96)

Commit:
Auto commit for SQL_Injection-a9050c65-76b3-4157-a83a-b3f98780f5f0

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/a9050c65-76b3-4157-a83a-b3f98780f5f0)

Commit:
Auto commit for SQL_Injection-33b4ac13-dd1d-4bff-a0c4-83c0ba92518b

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/33b4ac13-dd1d-4bff-a0c4-83c0ba92518b)

Commit:
Auto commit for SQL_Injection-c1b29106-33e0-4b58-bba7-422a4dfb1ede

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/c1b29106-33e0-4b58-bba7-422a4dfb1ede)

Commit:
Auto commit for SQL_Injection-ea3ebcb0-7b6c-4323-a449-54059d782dd9

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/ea3ebcb0-7b6c-4323-a449-54059d782dd9)

Commit:
Auto commit for SQL_Injection-54856ea4-4f62-4f26-8e04-60f7de51609b

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/54856ea4-4f62-4f26-8e04-60f7de51609b)

Commit:
Auto commit for SQL_Injection-144c0d78-19a5-4bfb-b8bf-a10c4a5c0e29

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/144c0d78-19a5-4bfb-b8bf-a10c4a5c0e29)

Commit:
Auto commit for SQL_Injection-c4c9e164-254f-4085-8d73-12d9e2040f03

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/c4c9e164-254f-4085-8d73-12d9e2040f03)

Commit:
Auto commit for SQL_Injection-edc18527-65c7-485c-bb51-1a07b5527237

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/edc18527-65c7-485c-bb51-1a07b5527237)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -45,7 +45,7 @@
         return (promise);
     },
     getTPDsByTPD: function (tpd) {
-        return tpdModel.findOne({ 'tld': tpd });
+        return tpdModel.findOne({ 'tld': mongoSanitize.sanitize({ data: tpd }).data });
     },
     getTPDsByWildcard: function (search, listOnly) {
         let promise;
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/edc18527-65c7-485c-bb51-1a07b5527237)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -36,7 +36,7 @@
     getTPDsByZone: function (zone, listOnly) {
         let promise;
         if (!listOnly) {
-            promise = tpdModel.find({ 'zones.zone': zone }).exec();
+            promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
                 'zones.zone': zone,
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/54856ea4-4f62-4f26-8e04-60f7de51609b)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -34,7 +34,7 @@
     getGraphLinksByZone: function (zone) {
         let limitQuery = { 'links': 1 };
         return graphLinksModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/ea3ebcb0-7b6c-4323-a449-54059d782dd9)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -34,7 +34,7 @@
     getGraphDataByZone: function (zone) {
         let limitQuery = { 'data': 1, 'errs': 1 };
         return graphDataModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/a9050c65-76b3-4157-a83a-b3f98780f5f0)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -31,7 +31,7 @@
     getGraphDocsByZone: function (zone) {
         let limitQuery = { 'docs': 1 };
         return graphDocsModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/144c0d78-19a5-4bfb-b8bf-a10c4a5c0e29)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -38,6 +38,6 @@
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
-        return (graphDataModel.countDocuments({ 'zone': zone }).exec());
+        return (graphDataModel.countDocuments({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec());
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/c4c9e164-254f-4085-8d73-12d9e2040f03)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -52,7 +52,7 @@
         let AWSregex = new RegExp('.*' + search + '$');
         if (listOnly) {
             promise = tpdModel.find({
-                'tld': { '$regex': AWSregex },
+                'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
             promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/33b4ac13-dd1d-4bff-a0c4-83c0ba92518b)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -31,6 +31,6 @@
 module.exports = {
     CertGraphModel: certGraphModel,
     getGraphDataByZone: function (zone) {
-        return certGraphModel.findOne({ 'zone': zone }).exec();
+        return certGraphModel.findOne({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/5b692a17-c2d3-42fb-b097-13d1c3013a96)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -34,7 +34,7 @@
     getGraphConfigByZone: function (zone) {
         let limitQuery = { 'config': 1 };
         return graphModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/29d20ca2-82f4-41c0-a7a2-ed7c7e20fe87/fix/c1b29106-33e0-4b58-bba7-422a4dfb1ede)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Commit:
Auto commit for SQL_Injection-d0b811ec-fc5d-4368-a984-75c8f10fda16

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/d0b811ec-fc5d-4368-a984-75c8f10fda16)

Commit:
Auto commit for SQL_Injection-e4489684-c2e5-4ad8-b0b7-65beece329de

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/e4489684-c2e5-4ad8-b0b7-65beece329de)

Commit:
Auto commit for SQL_Injection-8c2c8b16-7d06-484e-a0f1-fe0354630fd4

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/8c2c8b16-7d06-484e-a0f1-fe0354630fd4)

Commit:
Auto commit for SQL_Injection-57d88066-ea33-42fd-b3d0-7d57bd9071ee

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/57d88066-ea33-42fd-b3d0-7d57bd9071ee)

Commit:
Auto commit for SQL_Injection-38067ffb-4de1-4da2-a04a-e22b2c79f5e8

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/38067ffb-4de1-4da2-a04a-e22b2c79f5e8)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -40,7 +40,7 @@
             promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
-                'zones.zone': zone,
+                'zones.zone': mongoSanitize.sanitize({ data: zone }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         }
         return (promise);
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/38067ffb-4de1-4da2-a04a-e22b2c79f5e8)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphLinksModel: graphLinksModel,
     getGraphLinksByZone: function (zone) {
-        let limitQuery = { 'links': 1 };
+        let limitQuery = { 'links': mongoSanitize.sanitize({ data: 1 }).data };
         return graphLinksModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/57d88066-ea33-42fd-b3d0-7d57bd9071ee)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -30,7 +30,7 @@
 module.exports = {
     GraphDocsModel: graphDocsModel,
     getGraphDocsByZone: function (zone) {
-        let limitQuery = { 'docs': 1 };
+        let limitQuery = { 'docs': mongoSanitize.sanitize({ data: 1 }).data };
         return graphDocsModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/8c2c8b16-7d06-484e-a0f1-fe0354630fd4)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -56,7 +56,7 @@
                 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
-            promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
+            promise = tpdModel.find({ 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data }).exec();
         }
         return (promise);
     },
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/d0b811ec-fc5d-4368-a984-75c8f10fda16)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphModel: graphModel,
     getGraphConfigByZone: function (zone) {
-        let limitQuery = { 'config': 1 };
+        let limitQuery = { 'config': mongoSanitize.sanitize({ data: 1 }).data };
         return graphModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/a144e254-3284-4c93-9bf2-6ef075f722fb/fix/e4489684-c2e5-4ad8-b0b7-65beece329de)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.
